pytket/phir/rebasing/rebaser.py

```diff
@@ -15,4 +16,8 @@ def rebase_to_qtm_machine(circuit: Circuit, qtm_machine: str) -> Circuit:
     )
     # Optimization level 0 includes rebasing and little else
     # see: https://cqcl.github.io/pytket-quantinuum/api/#default-compilation
-    return backend.get_compiled_circuit(circuit, 0)
+    compiled = backend.get_compiled_circuit(circuit, 0)
+
+    # NOTE: Serializing and deserializing to remove global phase gates
+    as_qasm_str = circuit_to_qasm_str(compiled, header="hqslib1")
```

Comment:
After talking to the tket team, there appears some things we can do to avoid needing the roundtrip. We can first run the the `DecomposeBoxes` pass (https://cqcl.github.io/tket/pytket/api/passes.html) and then in sharding we can simply treat any operation whose primary command is the global phase gate as nop that doesn't have to be added as a shard.